src/app/SlaveInterface/test_SlaveInterface.c

```diff
@@ -0,0 +1,120 @@
+// includes for mocking
+#include "unity.h"
+
+// module under test
+#include "SlaveInterface.h"
+
+// mocked includes
+#include "MockFaultManager.h"
+#include "MockHAL_SlaveChips.h"
+
+/**
+ * The slave interface reads are basically memcpys. Make sure we didn't muck that up.
+ */
+void test_SlaveInterface_cell_info_read(void)
+{
+    BatteryModel_t bm;
+
+    // set up some voltages we expect to see in the same order in the battery model
+    float mock_voltages[NUM_SERIES_CELLS];
+    bool mock_drain_states[NUM_SERIES_CELLS];
+
+    for(int i = 0; i < NUM_SERIES_CELLS; i++)
+    {
+        mock_voltages[i] = i;
+        mock_drain_states[i] = (i % 2 == 0);
+    }
+
+    // first test with no error, new data should end up in battery model and faults should be cleared
+    Error_t err;
+    err.active = false;
+    HAL_SlaveChips_get_all_cell_data_ExpectAnyArgsAndReturn(err);
+    FaultManager_clear_fault_ExpectAnyArgs();
+    
+    HAL_SlaveChips_get_all_cell_data_ReturnArrayThruPtr_voltages(mock_voltages, NUM_SERIES_CELLS);
+    HAL_SlaveChips_get_all_cell_data_ReturnArrayThruPtr_is_draining(mock_drain_states, NUM_SERIES_CELLS);
+
+    SlaveInterface_read_cell_info(&bm);
+
+    // simultaneously check for copied data and copy over next data set
+    for(int i = 0; i < NUM_SERIES_CELLS; i++)
+    {
+        char err_msg[100];
+        sprintf(err_msg, "no error cell %d voltage not expected", i);
+        TEST_ASSERT_MESSAGE(bm.cells[i].voltage == i, err_msg);
+        sprintf(err_msg, "no error cell %d drain state not expected", i);
+        TEST_ASSERT_MESSAGE(bm.cells[i].is_draining == (i % 2 == 0), err_msg);
+        mock_voltages[i] = i * 10;
+        mock_drain_states[i] = (i % 2 == 1);
+    }
+
+    // now run again with an error. Modified voltages/drain states from for loop above should NOT appear in battery model
+    err.active = true;
```

Comment:
Should this be a separate test? Bc your basically testing two things in this single test

Reply:
I'm kindof testing two things, but setting all that up twice is a lot